pkg/kubectl/proxy/proxy_server.go

```diff
@@ -170,6 +172,13 @@ func makeUpgradeTransport(config *rest.Config) (proxy.UpgradeRequestRoundTripper
 	rt := utilnet.SetOldTransportDefaults(&http.Transport{
 		TLSClientConfig: tlsConfig,
 	})
+
+	rt.DialContext = (&net.Dialer{
```

Comment:
hrm, I don't think this is what you want. Instead, I think that you want:

```go
rt.Dial.Keepalive = keepalive
```

Reply:
@brendandburns Dial is deprecated, and DialContext id recommended. https://github.com/kubernetes/kubernetes/pull/60012 is making use of `DialContext`